src/amr/data/particles/particles_data.hpp

```diff
@@ -97,10 +105,20 @@ namespace amr
 
         using Particle_t          = typename ParticleArray::Particle_t;
         static constexpr auto dim = ParticleArray::dimension;
+        // without access to interp_order we cannot do this:
+        // static constexpr int ghostSafeMapLayer = ghostWidthForParticles<interp_order>() + 1;
+        // so we do that :
+        static constexpr int ghostSafeMapLayer = 4;
+        // which results in mapping more cells than strictly necessary but should never crash
 
     public:
         ParticlesData(SAMRAI::hier::Box const& box, SAMRAI::hier::IntVector const& ghost)
             : SAMRAI::hier::PatchData::PatchData(box, ghost)
+            , domainParticles{grow(phare_box_from<dim>(getGhostBox()), ghostSafeMapLayer)}
```

Comment:
is this like the ghostbox of the ghostbox? thinking

Reply:
not sure why I did this exactly.... 
so, the fact the ParticleData now needs a box is because of the cellmap not being a map anymore but a vector, so has to be preallocated to a size.... a size that is large enough for all cells possibly reached by particles...
I think this was added as a safety so that particles leaving the ghost box cells outward can still be updated in the cell map, 

I think the exact minimum value should be
        // static constexpr int ghostSafeMapLayer = ghostWidthForParticles<interp_order>() + 1;
as commented
but I don't have interporder here.... so hard coded 4 for all interporder........

Reply:
ah you mean it the safe layer could/should be only 1?

Reply:
yeah